### ISSUE TITLE
EntityBreakingFX should check item for color

### ISSUE THREAD

**Issue by kirderf1** [MinecraftForge/MinecraftForge#2018]
I've made an eatable and colored item (where the color is determined from the meta), only to find out that the particles created when eating is using the white, uncolored texture. My best try to fix this was to manually set the color when the entity is spawned into the world- But when I was trying that, I found out that the item and metadata wasn't kept by the particle, so I couldn't check if the particle was for my item, or which color it then should have.
I can think of three different approaches:
*Make the ITEM_CRACK particle factory use item.getColorFromItemStack()
*Make EntityBreakingFX store the Item and metadata in variables to let mods view them.
*Add an event at an appropriate place that allows view of item + metadata and the possibility to change the color of the particle (EntityFX.setRBGColorF())
Edit: Changed EntityBreakingFX to related factory
TL;DR: I want support for using custom color using item.getColorFromItemStack() for food items, as the food particles doesn't care about any custom color.

**Comment by clienthax** [comment]
I believe it colours it if the colour multiplier is set, (which doesn't colour the block for some reason)

**Comment by kirderf1** [comment]
You mean the particle.setRBGColorF() method? The problem is that I can't find a way of checking "if that pacticle is for the colored item" and which color it should have. Normally, I think that method is to be used after creation by the creator, but the code for creating item breaking particles is outside my code, so I can't do anything about that.

**Comment by clienthax** [comment]
No in the block class there's a method that returns a into called colorMultiplier, if you override it and return a rgb value it should colour the particles

**Comment by clienthax** [comment]
Aha I completely misred this, sorry

**Comment by kirderf1** [comment]
Looking at the latest version of forge, this is still a problem. The particle in question has since been renamed to `ParticleBreaking` (and `EntityFX` to `Particle`), but the situation appears to otherwise still be according to above.

**Comment by matthew-jumper** [comment]
I will take a look at this issue and see if it can be resolved.
